transmart-core-api/src/main/groovy/org/transmartproject/core/multidimquery/MultiDimensionalDataResource.groovy

```diff
@@ -102,7 +102,20 @@ interface MultiDimensionalDataResource {
 
     QueryResult createPatientSetQueryResult(String name, MultiDimConstraint constraint, User user, String constraintText, String apiVersion)
 
-    QueryResult findQueryResult(Long queryResultId, User user)
+    /**
+     * Update existing patient set that satisfies provided constraints
+     * @param name
+     * @param constraint
+     * @param user
+     * @param constraintText
+     * @param apiVersion
+     * @return A patient set with updated results (patient ids)
+     */
+    QueryResult updatePatientSetQueryResult(String name, MultiDimConstraint constraint, User user, String constraintText)
```

Comment:
Seems that `constraint` and `constraintText` are duplicates. If it is an existing patient set (or query), also the id of the set/query can be used.